python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -1,9 +1,30 @@
-import contextlib
-from typing import Iterator
+from contextlib import AbstractContextManager, AbstractAsyncContextManager
+from typing import Dict, Type, Sequence, List, Any
+from openinference.semconv.trace import SpanAttributes
+from opentelemetry.context import (
+    _SUPPRESS_INSTRUMENTATION_KEY,
+    attach,
+    detach,
+    set_value,
+    get_value,
+    get_current,
+)
+from opentelemetry.util.types import AttributeValue
+from typing import (
+    Iterator,
+    Tuple,
+)
 
+CONTEXT_ATTRIBUTES = (
+    SpanAttributes.SESSION_ID,
+    SpanAttributes.USER_ID,
+    SpanAttributes.METADATA,
+    SpanAttributes.TAG_TAGS,
+)
 
-@contextlib.contextmanager
-def suppress_tracing() -> Iterator[None]:
+
+class suppress_tracing(AbstractContextManager):
+    slots = ["_token"]
```

Comment:
Did you intend to use `__slots__` here? Not sure we need `__slots__` tbh. It is a minor optimization and reduces readability IMO.

Reply:
Removed